app/actions/views/user_profile.js

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {getDirectChannelName} from 'service/utils/channel_utils';
+import {createDirectChannel} from 'service/actions/channels';
+import {getTeamMember} from 'service/actions/teams';
+import {handleSelectChannel, toggleDMChannel} from 'app/actions/views/channel';
+import {NavigationTypes} from 'app/constants';
+
+export function makeDirectChannel(otherUserId) {
```

Comment:
yeah that's the one I copied, wanted to keep them separate since the profile action has some additional logic for navigating back to the channel. I can handle it another way to where we don't duplicate the code.